pubsub/adapters/googlecloud.py

```diff
@@ -1,52 +1,87 @@
-from google.cloud import pubsub
+import logging
+
+from google.cloud import pubsub_v1
+from google.gax.errors import GaxError
+from grpc import StatusCode
+
 from pubsub.adapters.base import BaseAdapter
 from pubsub.adapters.exceptions import IdentifierRequiredException, TopicNotFound, SubscriptionNotFound
 
+log = logging.getLogger('pubsub')
+
 
 class GooglePubsub(BaseAdapter):
     """
     Google-cloud adapter class
     """
 
-    def __init__(self, client_identifier='default'):
-        self.pubsub_client = pubsub.Client()
+    def __init__(self, project_id, client_identifier='default'):
+        self.publisher = pubsub_v1.PublisherClient()
+        self.subscriber = pubsub_v1.SubscriberClient()
         self.client_identifier = client_identifier
+        self.project_id = project_id
+
+    def ack(self, message):
+        message.ack()
 
     def publish(self, topic_name, message):
-        topic = self.pubsub_client.topic(topic_name)
-        if not topic.exists():
-            topic.create()
-        topic.publish(message)
+        topic_path = self.publisher.topic_path(self.project_id, topic_name)
+        exists = True
+        try:
+            self.publisher.get_topic(topic_path)
+        except GaxError as exc:
+            if exc.cause._state.code != StatusCode.NOT_FOUND:
+                raise
+            exists = False
```

Comment:
Can we not do `self.publisher.create_topic(topic_path)` here without the need for the boolean exists?

Reply:
👍